packages/cf-component-table/src/TableCell.js

```diff
@@ -1,37 +1,186 @@
+      ? 0
+      : undefined,
 
-    if (className && className.trim()) {
-      _className += ' ' + className.trim();
+    // Responsive styles
+    [`@media (max-width: ${theme.breakpoints.tablet})`]: {
+      display: 'block',
+      width: '100%',
+      boxSizing: 'border-box',
+      borderTopWidth: 0,
+      borderTop: cellIndex === 0
+        ? `1px solid ${theme.tableBordercolor}`
+        : undefined
+    },
+
+    // Row Styles
+    borderColor: theme[`table${capitalize(rowType)}BorderColor`],
+    backgroundColor: striped && Math.abs(rowIndex) % 2 === 1
+      ? theme[`tableStriped${capitalize(rowType)}Color`]
+      : theme[`table${capitalize(rowType)}Color`],
+
+    ':hover': hover
+      ? { backgroundColor: theme[`tableHover${capitalize(rowType)}Color`] }
+      : undefined,
+    borderLeft: bordered ? `1px solid ${theme.tableBorderColor}` : undefined,
+
+    [`@media screen and (max-width: ${theme.breakpoints.tablet})`]: {
+      display: 'block',
+      border: 'none',
+      float: 'left',
+      clear: 'left',
+      padding: 0,
+
+      // Mobile stripe issue fix
+      backgroundColor: striped && Math.abs(rowIndex) % 2 === 1
+        ? 'transparent'
+        : undefined
     }
+  });
 
-    return (
-      <td className={_className} {...props}>
-        {this.props.children}
-      </td>
+class TableCell extends React.Component {
+  render() {
+    const {
+      is,
+      className,
+      children,
+      condensed,
+      striped,
+      bordered,
+      hover,
+      bare,
+      rowType,
+      theadIndex,
+      tbodyIndex,
+      tfootIndex,
+      rowIndex,
+      cellIndex,
+      align,
+      width,
+      rowSpan,
+      colSpan
+    } = this.props;
+
+    return React.createElement(
+      is,
+      {
+        className,
+        rowSpan: is === 'div' ? undefined : rowSpan,
+        colSpan: is === 'div' ? undefined : colSpan
+      },
+      this.props.children
     );
   }
 }
 
 TableCell.propTypes = {
-  className: PropTypes.string,
+  is: PropTypes.oneOf(['td', 'div']),
+  className: PropTypes.string.isRequired,
   align: PropTypes.oneOf(['left', 'center', 'right']),
-  children: PropTypes.node
+  rowSpan: PropTypes.number,
+  colSpan: PropTypes.number,
+  headers: PropTypes.arrayOf(PropTypes.string),
+  children: PropTypes.node,
+  // implicit style props
+  condensed: PropTypes.bool,
+  striped: PropTypes.bool,
+  bordered: PropTypes.bool,
+  hover: PropTypes.bool,
+  bare: PropTypes.bool,
+  rowType: PropTypes.string,
+  theadIndex: PropTypes.number,
+  tbodyIndex: PropTypes.number,
+  tfootIndex: PropTypes.number,
+  rowIndex: PropTypes.number,
+  cellIndex: PropTypes.number,
+  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
 };
 
 TableCell.defaultProps = {
-  className: ''
+  is: 'td',
+  align: 'left'
 };
 
-export default TableCell;
+export const createTableCell = (...styles) =>
+  applyTheme(connectStyles(TableCellStyles, ...styles)(TableCell), TableTheme);
```

Comment:
You do get access to it, we are using our wrapped `createComponent`, which will automatically passThru props if defined in propTypes.